Testscripts/Windows/HB-CLOCKSYNC.ps1

```diff
@@ -0,0 +1,305 @@
+		Start-Sleep -seconds 1
+
+		#Verify the VM status after power on event
+		$vmStatus = Get-AzVM -Name $vmName -ResourceGroupName $rgName -Status
+		if ($vmStatus.Statuses[1].DisplayStatus -eq "VM running") {
+			Write-LogInfo "$($vmStatus.Statuses[1].DisplayStatus): Verified successfully VM status is running after resuming"
+		} else {
+			Write-LogErr "$($vmStatus.Statuses[1].DisplayStatus): Could not find the VM status after resuming"
+			throw "Can not identify VM status after resuming"
+		}
+
+		# Verify the kernel panic, call trace or fatal error
+		$calltrace_filter = Run-LinuxCmd -ip $AllVMData[0].PublicIP -port $AllVMData[0].SSHPort -username $user -password $password -command "dmesg | grep -iE '(call trace|fatal error)'" -ignoreLinuxExitCode:$true
+
+		if ($calltrace_filter -ne "") {
+			Write-LogErr "Found Call Trace or Fatal error in dmesg"
+			# The throw statement is commented out because this is linux-next, so there is high chance to get call trace from other issue. For now, only print the error.
+			# throw "Call trace in dmesg"
+		} else {
+			Write-LogInfo "Not found Call Trace and Fatal error in dmesg"
+		}
+
+		Write-LogDbg "Waiting 60-second for logging sync"
+		Start-Sleep -seconds 60
+
+		# Check the system log if it shows Power Management log
+		"hibernation entry", "hibernation exit" | ForEach-Object {
+			$pm_log_filter = Run-LinuxCmd -ip $AllVMData[0].PublicIP -port $AllVMData[0].SSHPort -username $user -password $password -command "cat /var/log/syslog | grep -i '$_'" -ignoreLinuxExitCode:$true
+			Write-LogInfo "Searching the keyword: $_"
+			if ($pm_log_filter -eq "") {
+				Write-LogErr "Could not find Power Management log in dmesg"
+				throw "Missing PM logging in dmesg"
+			} else {
+				Write-LogInfo "Successfully found Power Management log in dmesg"
+				Write-LogInfo $pm_log_filter
+			}
+		}
+
+		Copy-RemoteFiles -downloadFrom $AllVMData[0].PublicIP -port $AllVMData[0].SSHPort -username $user -password $password -download -downloadTo $LogDir -files "*.log" -runAsSudo
+
+		$beforeTimeStamp = Get-Content $LogDir\before_timestamp.log
+		Write-LogDbg $beforeTimeStamp
+		$12minAfterTimeStamp = Get-Content $LogDir\12min_after_timestamp.log
+		Write-LogDbg $12minAfterTimeStamp
+
+		if ($beforeTimeStamp -ne $12minAfterTimeStamp) {
+			Write-LogInfo "Successfully verified the beforeTimeStamp was different from 12minAfterTimeStamp"
+		} else {
+			Write-LogErr "Did not find time synced. $beforeTimeStamp to $12minAfterTimeStamp"
+		}
+
+		$controllerTimeStamp = Get-Date -format "yyyy"
+		Write-LogDbg $controllerTimeStamp
+
+		if ($12minAfterTimeStamp -eq $controllerTimeStamp) {
+			Write-LogInfo "Successfully verified the system date changed back to correct date"
+		} else {
+			Write-LogErr "Expected VM time changed back to the correct one after sync-up, but found $12minAfterTimeStamp"
```

Comment:
when hit the error, the test should fail?

Reply:
Fixed in 3d27b72ed5f302940df8a778f189649817382860